pywt/_extensions/_swt.pyx

```diff
@@ -201,8 +202,10 @@ cpdef swt_axis(np.ndarray data, Wavelet wavelet, size_t level,
         cA = np.empty(output_shape, dtype=data.dtype)
         cD = np.empty(output_shape, dtype=data.dtype)
         # strides won't match data_info.strides if data is not C-contiguous
-        output_info.strides = <pywt_index_t *> cA.strides
-        output_info.shape = <size_t *> cA.shape
+        strides_view = <signed long [:data.ndim]> <signed long *> cA.strides
+        strides_view = strides_view.copy()
+        output_info.strides = <pywt_index_t *> &strides_view[0]
+        output_info.shape = <size_t *> &output_shape[0]
```

Comment:
A copy was required here, since cA would go out of scope after redefinition on any of the complex cases

Reply:
Good catch! That would explain a crash indeed. I'm not quite sure whether this isn't happening also further down, since there is another pattern like that where an ndarray or memory view is overwritten. It's probably protected only by this code that keeps a reference:
```python
        if not trim_approx:
            ret.append((cA, cD))
```
however, the default for `trim_approx` is False. It all looks a bit fragile.

Reply:
Did you audit the other `.pyx` files for this pattern?

Reply:
> Did you audit the other `.pyx` files for this pattern?

I did this audit, and didn't find a pattern like in this file with an ndarray being overwritten and then being reused in a for-loop.